Option to specify bower path

leMaik opened the pull request:
For some project, the bower directory may not be named `bower_components` but rather `vendor`. It might also be in a different directory specified in `bower.rc`.
To make bower-requirejs-auto work in this case, I added a config node to specify the bower path, relative to the base path. See README.md for documentation.

The second commit is strange... I took the code from the bower package of this project and it was there but not in this git.

leMaik committed:
Add option to configure bower path.

leMaik committed:
Ensure that required file is a .js file.

leMaik committed:
Update README.

AndersDJohnson commented:
@leMaik - Thanks for the PR. This is a good feature you've added. I'll merge it in.

I wonder if it would be even better to also support automatically determining a custom `bower_components` directory by making an AJAX request to check for `.bowerrc` file, then using its `"directory"` value, if defined. Are you interested in implementing that?

Also, I see your JS restriction change. I know my original code sort of assumed that this would only apply to JS files, so it's understandable you'd build on that. But there are use cases with RequireJS plugins where you might want to load CSS, TXT, and other file types. So I think this could be refactored at some point to support those.
Looks like I was iterating through main files and whichever file comes last (now whichever JS file, thanks to you change) gets to clobber the RequireJS paths registration. Instead, we could consider registering multiple RequireJS modules when a Bower module exposes more than one main file, e.g. "my-module/index" (JS), "my-module/helper" (JS), "my-module/style" (CSS), etc.
Thoughts? Did you have an example Bower module you were dealing with that made this change necessary?

leMaik commented:
@adjohnson916  - I thought about the AJAX request variant too. But the problem was that grunt-live-server (which we use) could not access it as it is located outside the directory that it serves, so I dropped that and just added the new attribute.

Registering multiple files might break a lot of packages because you'd first register their `.js` file and then their `.min.js` file, which expose the same components. I wonder how we could solve that, maybe by preferring and only registering one file of each name? The `.js` would be more useful for debugging purposes. A more powerful way would be to offer an optional selector function so that the user could configure which modules are required, if there are problems. I'd say that this is only needed in very few cases (I have no bower module that requires this, unfortunately).

The selector function could be something like this:

```
function (module, files) {
   if (module == 'some-strange-module) {
      //only expose the first to modules, as the other cause problems
      return [files[0], files[1]];
   }
   return files; //don't change file list to require for other modules
}
```

AndersDJohnson commented:
@leMaik - Thanks, good thoughts. Please star this repo to let me know you're interested in further enhancements!

leMaik commented:
Would it be possible to update the bower package so that I don't have to use the specific commit version anymore? :) This would also help us getting future updates automatically using bower, again.

AndersDJohnson commented:
@leMaik, I've released as https://github.com/adjohnson916/bower-requirejs-auto/releases/tag/v0.1.0. Please let me know if it doesn't work!
